mattermost-plugin/webapp/src/components/rhsChannelBoards.tsx

```diff
@@ -41,10 +41,16 @@ const RHSChannelBoards = () => {
     const me = useAppSelector<IUser|null>(getMe)
     const dispatch = useAppDispatch()
     const intl = useIntl()
+    const [dataLoaded, setDataLoaded] = React.useState(false)
 
     useEffect(() => {
-        dispatch(loadBoards())
-        dispatch(fetchMe())
+        async function loadRhsRequiredData() {
+            await dispatch(loadBoards())
```

Comment:
I would prefer here to use something different, something like an `await Promise.all(...)` and then the `setDataLoaded(true)`. Also, I don't love creating an asyncrhonous function just to make it async. you can still use something like:

```ts
useEffect(() => {
    Promise.all([dispatch(loadBoards()), dispatch(loadMyBoardsMemberships()),  dispatch(fetchMe())]).then(() => setDataLoaded(true))
}, [])
```

Reply:
ok, thanks for the feedback!